common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -482,6 +482,8 @@ def do_connection(
 
         self._course_run_cache = {}
 
+        self.detached_categories = [name for name, __ in XBlock.load_tagged_classes("detached")]
```

Comment:
You may as well make this a global constant, since the list is set at install-time.